safe/report/processors/default.py

```diff
@@ -268,25 +268,29 @@ def qgis_composer_renderer(impact_report, component):
         item_id = map_el.get('id')
         split_count = map_el.get('grid_split_count')
         layers = map_el.get('layers')
+        map_extent_option = map_el.get('extent')
         composer_map = composition.getComposerItemById(item_id)
         """:type: qgis.core.QgsComposerMap"""
         if composer_map:
             composer_map.setKeepLayerSet(True)
             composer_map.setLayerSet(
                 [l.id() for l in layers])
-            # composer_map.zoomToExtent(square_extent)
-            extent = QgsRectangle()
-            extent.setMinimal()
-            for l in layers:
-                extent.combineExtentWith(l.extent())
-
-            canvas_extent = extent
-            width = canvas_extent.width()
-            height = canvas_extent.height()
+            if map_extent_option and isinstance(
+                    map_extent_option, QgsRectangle):
+                extent = map_extent_option
+            else:
+                extent = QgsRectangle()
+                extent.setMinimal()
+                for l in layers:
+                    # combine extent if different layer is provided.
+                    extent.combineExtentWith(l.extent())
```

Comment:
This is if the extent were not passed by extractor, for example if we use different extractor other than default (which pass analysis_extent as default). Then the map needs to find some extent to show as default, so I combined all extent for all layers passed by extractor.
For the dock use case, it will not reach the else part. This is only useful for realtime or headless where it needs custom extractor (not inasafe default).